e2e/local_repository/local_repository_test.go

```diff
@@ -0,0 +1,110 @@
+package simple
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/bazelbuild/bazel-watcher/e2e"
+	"github.com/bazelbuild/rules_go/go/tools/bazel_testing"
+)
+
+const secondaryBuild = `
+sh_library(
+	name = "lib",
+	data = ["lib.sh"],
+	visibility = ["//visibility:public"],
+)
+`
+
+const secondaryLib = `
+function say_hello {
+	printf "hello!"
+}
+`
+
+const secondaryLibAlt = `
+function say_hello {
+	printf "hello2!"
+}
+`
+
+const mainFiles = `
+-- BUILD.bazel --
+sh_binary(
+	name = "test",
+	srcs = ["test.sh"],
+	deps = [
+		"@secondary//:lib",
+	],
+)
+-- test.sh --
+#!/bin/bash
+source ../secondary/lib.sh
+say_hello
+-- WORKSPACE --
+local_repository(
+    name = "secondary",
+    path = "../secondary",
+)
+`
+
+var (
+	secondaryWd string
+)
+
+func TestMain(m *testing.M) {
+	bazel_testing.TestMain(m, bazel_testing.Args{
+		Main: mainFiles,
+		SetUp: func() error {
+			// Create a secondary workspace in a sibling folder of the main workspace.
+			secondaryWd, _ = filepath.Abs(filepath.Join("..", "secondary"))
+
+			// Manually create files in the secondary workspace.
+			os.Mkdir(secondaryWd, 0777)
```

Comment:
You should check the exit code of these functions and return error if they fail.